ddtrace/contrib/unittest/patch.py

```diff
@@ -423,13 +529,63 @@ def handle_test_wrapper(func, instance, args: tuple, kwargs: dict):
-            _CIVisibility._unittest_data["suites"][test_module_suite_path]["suite_span"] = test_suite_span
+            suite_dict = _CIVisibility._unittest_data["suites"][test_module_suite_path]
+            suite_dict["suite_span"] = test_suite_span
         if not test_module_span or not test_suite_span:
             log.debug("Suite and/or module span not found for test: %s", test_name)
             return func(*args, **kwargs)
         with _start_test_span(instance, test_suite_span) as span:
-            result = func(*args, **kwargs)
+            test_session_span = _CIVisibility._datadog_session_span
+            root_directory = os.getcwd()
+            if _CIVisibility.test_skipping_enabled():
+                if _is_marked_as_unskippable(instance):
+                    span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_module_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                    test_session_span.set_tag_str(test.ITR_UNSKIPPABLE, "true")
+                test_name = _extract_test_method_name(instance)
+                test_module_suite_path_without_extension = "{}/{}".format(
+                    os.path.splitext(test_module_path)[0], test_suite_name
+                )
+                if (
+                    len(args)
+                    and _CIVisibility._instance._should_skip_path(test_module_suite_path_without_extension, test_name)
+                    and not _is_skipped_test(instance)
+                ):
+                    global _global_skipped_elements
+                    _global_skipped_elements += 1
+
+                    test_module_span._metrics[test.ITR_TEST_SKIPPING_COUNT] += 1
+                    test_module_span.set_tag_str(test.ITR_TEST_SKIPPING_TESTS_SKIPPED, "true")
+                    test_module_span.set_tag_str(test.ITR_DD_CI_ITR_TESTS_SKIPPED, "true")
+
+                    test_session_span.set_tag_str(test.ITR_TEST_SKIPPING_TESTS_SKIPPED, "true")
+                    test_session_span.set_tag_str(test.ITR_DD_CI_ITR_TESTS_SKIPPED, "true")
+
+                    if _is_marked_as_unskippable(instance):
+                        span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_module_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                        test_session_span.set_tag_str(test.ITR_FORCED_RUN, "true")
+                    else:
+                        instance._dd_itr_skip = True
+                        span.set_tag_str(test.ITR_SKIPPED, "true")
+                        span.set_tag_str(test.SKIP_REASON, SKIPPED_BY_ITR_REASON)
+
+            if _is_test_coverage_enabled(instance):
+                coverage = _start_coverage(root_directory)
+                instance._coverage = coverage
+            if is_skipped_by_itr(instance):
+                result = args[0]
+                result.startTest(test=instance)
+                result.addSkip(test=instance, reason=SKIPPED_BY_ITR_REASON)
```

Comment:
Do you know when `result` is ever `None` ? [TestCase.run()](https://github.com/python/cpython/blob/main/Lib/unittest/case.py#L599) and [its docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.run) don't say much.

I'm wondering if you also need to call [TestResult.stopTest()](https://docs.python.org/3/library/unittest.html#unittest.TestResult.stopTest) if you're going to manually run the test.

I think I like the way we do it in the `pytest` plugin where we just add the skip marker (with our reason) and then still let `pytest` "run" the test as if it were a regular one. It makes me feel a little less paranoid that we might forget to do something the underlying runner does on its own.